superset/assets/javascripts/SqlLab/components/SaveQuery.jsx

```diff
@@ -0,0 +1,139 @@
+/* global notify */
+import React from 'react';
+import { FormControl, FormGroup, Overlay, Popover, Row, Col } from 'react-bootstrap';
+import Button from '../../components/Button';
+const $ = window.$ = require('jquery');
+
+const propTypes = {
+  defaultLabel: React.PropTypes.string,
+  sql: React.PropTypes.string,
+  schema: React.PropTypes.string,
+  dbId: React.PropTypes.number,
+  animation: React.PropTypes.bool,
+};
+const defaultProps = {
+  defaultLabel: 'Undefined',
+  animation: true,
+};
+
+class SaveQuery extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      description: '',
+      label: props.defaultLabel,
+      showSave: false,
+    };
+    this.toggleSave = this.toggleSave.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.onCancel = this.onCancel.bind(this);
+    this.onLabelChange = this.onLabelChange.bind(this);
+    this.onDescriptionChange = this.onDescriptionChange.bind(this);
+  }
+  onSave() {
+    const url = '/savedqueryviewapi/api/create';
+    const data = {
+      label: this.state.label,
+      description: this.state.description,
+      db_id: this.props.dbId,
+      schema: this.props.schema,
+      sql: this.props.sql,
+      csrf_token: $('input#csrf_token').val(),
```

Comment:
i think a better pattern here might be to send this as a header on the xhr request.
http://flask-wtf.readthedocs.io/en/stable/csrf.html#javascript-requests